api/v0/finder/client/http/client.go

```diff
@@ -37,37 +41,161 @@ func New(baseURL string, options ...httpclient.Option) (*Client, error) {
 	baseURL = u.String()
 	return &Client{
 		c:            c,
+		keyer:        indexer.NewKeyer(),
 		finderURL:    baseURL + finderPath,
 		providersURL: baseURL + providersPath,
 		statsURL:     baseURL + statsPath,
 	}, nil
 }
 
 // Find queries indexer entries for a multihash
-func (c *Client) Find(ctx context.Context, m multihash.Multihash) (*model.FindResponse, error) {
-	u := fmt.Sprint(c.finderURL, "/", m.B58String())
+func (c *Client) Find(ctx context.Context, mh multihash.Multihash) (*model.FindResponse, error) {
+
+	// query value keys from indexer
+	smh, err := dhash.SecondMultihash(mh)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprint(c.finderURL, "/", smh.B58String())
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u, nil)
 	if err != nil {
 		return nil, err
 	}
+	req.Header.Add("Accept", "application/json")
+
+	resp, err := c.c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
```

Comment:
Return here does not close `resp.Body`. Suggest adding `defer resp.Body.Close()` above, after getting the response.